lib/amqp_client.js

```diff
@@ -285,12 +285,13 @@ AMQPClient.prototype.disconnect = function() {
     debug('disconnecting');
     if (self._connection) {
       self._preventReconnect();
-      self._connection.on(Connection.Disconnected, function() {
-        self._connection = null;
+      var connection = self._connection;
+      self._clearConnectionState();
+      connection.once(Connection.Disconnected, function() {
+        debug('disconnected');
```

Comment:
`connection = undefined` in here? You've transferred the connection object over to the local reference to it, but I'm not sure if the reference is guaranteed to be cleaned up (though atm I can't see why it wouldn't be).. better to be safe?